rskj-core/src/main/java/org/ethereum/rpc/JsonRpcNettyServer.java

```diff
@@ -0,0 +1,72 @@
+package org.ethereum.rpc;
+
+import co.rsk.rpc.CorsConfiguration;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.handler.codec.http.*;
+import io.netty.handler.codec.http.cors.CorsConfig;
+import io.netty.handler.codec.http.cors.CorsHandler;
+import io.netty.handler.logging.LogLevel;
+import io.netty.handler.logging.LoggingHandler;
+
+public class JsonRpcNettyServer {
+
+    private int port;
+    private final EventLoopGroup bossGroup;
+    private final EventLoopGroup workerGroup;
+    private int socketLinger;
+    private boolean reuseAddress;
+    private CorsConfiguration corsConfiguration;
+    private final JsonRpcWeb3ServerHandler jsonRpcWeb3ServerHandler;
+
+    public JsonRpcNettyServer(int port, int socketLinger, boolean reuseAddress, CorsConfiguration corsConfiguration, JsonRpcWeb3ServerHandler jsonRpcWeb3ServerHandler) {
+        this.port = port;
+        this.socketLinger = socketLinger;
+        this.reuseAddress = reuseAddress;
+        this.corsConfiguration = corsConfiguration;
+        this.jsonRpcWeb3ServerHandler = jsonRpcWeb3ServerHandler;
+        this.bossGroup = new NioEventLoopGroup();
+        this.workerGroup = new NioEventLoopGroup();
+    }
+
+    public void start() throws InterruptedException {
+        ServerBootstrap b = new ServerBootstrap();
+        b.option(ChannelOption.SO_LINGER, socketLinger);
+        b.option(ChannelOption.SO_REUSEADDR, reuseAddress);
+        b.group(bossGroup, workerGroup)
+            .channel(NioServerSocketChannel.class)
+            .handler(new LoggingHandler(LogLevel.INFO))
+            .childHandler(new ChannelInitializer<SocketChannel>() {
+                @Override
+                protected void initChannel(SocketChannel ch) throws Exception {
+                    ChannelPipeline p = ch.pipeline();
+                    p.addLast(new HttpRequestDecoder());
+                    p.addLast(new HttpObjectAggregator(1024 * 1024 * 5));
+                    p.addLast(new HttpResponseEncoder());
+                    p.addLast(new HttpContentCompressor());
+                    if (corsConfiguration.hasHeader()) {
+                        p.addLast(new CorsHandler(
+                            CorsConfig
+                                .withOrigin(corsConfiguration.getHeader())
+                                .allowedRequestHeaders(HttpHeaders.Names.CONTENT_TYPE)
+                                .allowedRequestMethods(HttpMethod.POST)
```

Comment:
* Content-type header is allowed but the type is not enforced (application/json).
 * Previous code allowed GET and OPTIONS
 * Previous code added Vary: Origin header to avoid caching problems
 * Previous code set Access-Control-Max-Age to 600